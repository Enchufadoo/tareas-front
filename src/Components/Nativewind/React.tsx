import { styled as nativewindStyled } from 'nativewind'
import * as ReactNative from 'react-native'
import * as VectorIcons from '@expo/vector-icons'
import { styled } from 'dripsy'

export const Text = styled(nativewindStyled(ReactNative.Text))({
  color: '$text'
})

export const View = styled(nativewindStyled(ReactNative.View))()

export const Pressable = styled(nativewindStyled(ReactNative.Pressable))()
export const TextInput = styled(nativewindStyled(ReactNative.TextInput))()
export const TouchableOpacity = styled(
  nativewindStyled(ReactNative.TouchableOpacity)
)()
export const Switch = styled(nativewindStyled(ReactNative.Switch))()
export const SimpleLineIcons = styled(
  nativewindStyled(VectorIcons.SimpleLineIcons)
)()
export const MaterialIcons = styled(
  nativewindStyled(VectorIcons.MaterialIcons)
)()

export const Entypo = styled(nativewindStyled(VectorIcons.Entypo))()

export const Octicons = styled(nativewindStyled(VectorIcons.Octicons))()
