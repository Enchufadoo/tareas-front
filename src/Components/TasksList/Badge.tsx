import React from 'react'
import { Text, View } from '../Nativewind/React'

type Props = {
  count: number
  selected: boolean
} & React.ComponentPropsWithoutRef<typeof View>

const Badge = (props: Props) => {
  return (
    <View
      className="justify-center flex-1"
      {...props}
      sx={{ maxWidth: 24, maxHeight: 24 }}
    >
      <View
        sx={{
          borderColor: props.selected ? '$primary' : '$text'
        }}
        className={'rounded-xl  border flex-1  items-center justify-center'}
      >
        <Text
          sx={{
            color: props.selected ? '$primary' : '$text'
          }}
        >
          {props.count}
        </Text>
      </View>
    </View>
  )
}

export default Badge
