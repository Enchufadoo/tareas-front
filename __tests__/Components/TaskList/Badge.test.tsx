import React from 'react'
import { screen } from '@testing-library/react-native'
import Badge from '@/Components/TasksList/Badge'
import { renderWithProviders } from '@/Testing/TestUtil'

describe('Badge', () => {
  it('renders correctly when selected', () => {
    renderWithProviders(<Badge count={3} selected={true} />)
    const countElement = screen.getByText('3')

    expect(countElement).toBeTruthy()
  })

  it('renders correctly when not selected', () => {
    renderWithProviders(<Badge count={3} selected={false} />)
    const countElement = screen.getByText('3')
    expect(countElement).toBeTruthy()
  })
})
