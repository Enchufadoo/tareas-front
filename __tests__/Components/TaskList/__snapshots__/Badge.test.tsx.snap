// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge renders correctly 1`] = `
<View
  count={3}
  selected={true}
  style={
    [
      [
        {
          "justifyContent": "center",
        },
      ],
    ]
  }
>
  <View
    style={
      [
        [
          {
            "borderBottomLeftRadius": 12,
            "borderBottomRightRadius": 12,
            "borderTopLeftRadius": 12,
            "borderTopRightRadius": 12,
          },
          {
            "color": "#fff",
          },
          {
            "justifyContent": "center",
          },
          {
            "marginRight": 8,
          },
          {
            "borderBottomWidth": 1,
            "borderLeftWidth": 1,
            "borderRightWidth": 1,
            "borderTopWidth": 1,
          },
          {
            "borderBottomColor": "#5c5de5",
            "borderLeftColor": "#5c5de5",
            "borderRightColor": "#5c5de5",
            "borderTopColor": "#5c5de5",
          },
          {
            "height": 22,
            "paddingHorizontal": 8,
          },
        ],
      ]
    }
  >
    <Text
      style={
        [
          [
            {
              "fontSize": 14,
              "lineHeight": 20,
            },
            {
              "color": "#000",
            },
            {
              "color": "#5c5de5",
            },
            {
              "fontSize": 10,
            },
          ],
        ]
      }
    >
      3
    </Text>
  </View>
</View>
`;
